extra/src/main/java/aima/extra/search/uninformed/IterativeDeepeningAlphaBetaSearch.java

```diff
@@ -0,0 +1,120 @@
+package aima.extra.search.uninformed;
+
+import aima.core.search.api.Game;
+import aima.core.search.api.SearchForAdversarialActionFunction;
+
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+/**
+ * @author manthan.
+ */
+public class IterativeDeepeningAlphaBetaSearch<A, S, P> implements SearchForAdversarialActionFunction<S, A> {
```

Comment:
Could you order <A, S, P> the same as AlphaBetaSearch i.e. <S, A, P> and add @param descriptions to the class JavaDoc in a similar way.